### PR TITLE
Convert documentation to puppet-strings

### PR DESCRIPTION
**Pull request by ekohl**
While this isn't perfect, it's a huge step in the direction.

### CODE REVIEW

File: manifests/resource/mailhost.pp [hunk 1]
```diff
@@ -102,10 +153,10 @@
   Optional[String] $ssl_trusted_cert             = undef,
   Optional[Integer] $ssl_verify_depth            = undef,
   Enum['on', 'off', 'only'] $starttls            = 'off',
-  $protocol                                      = undef,
+  Optional[Enum['imap', 'pop3', 'smtp']] $protocol = undef,
```
Comment (ekohl) [inline manifests/resource/mailhost.pp:156]: These are the only values allowed. Previously only documented but https://nginx.org/en/docs/mail/ngx_mail_core_module.html#protocol agrees.

File: manifests/resource/mailhost.pp [hunk 2]
```diff
@@ -102,10 +153,10 @@
...
   Optional[String] $auth_http                    = undef,
   Optional[String] $auth_http_header             = undef,
-  String $xclient                                = 'on',
+  Enum['on', 'off'] $xclient                     = 'on',
```
Comment (ekohl) [inline manifests/resource/mailhost.pp:159]: Same here: previously documented and docs agree: https://nginx.org/en/docs/mail/ngx_mail_proxy_module.html#xclient

**Review by bastelfreak — APPROVED** [review]

**Comment by ekohl** [comment]
Updated, didn't realize there was some test passing in an invalid value.

**Comment by ekohl** [comment]
Should have read more carefully, the other one was exactly the same situation.

**Comment by ekohl** [comment]
I want to keep the parameter types out and leave that to https://github.com/voxpupuli/puppet-nginx/pull/1413, will update later.

**Review by ghoneycutt — APPROVED** [review]
Awesome work @ekohl !!

Have you tested generating a REFERENCE.md?

**Comment by ekohl** [comment]
Rebased to resolve conflicts. I did generate one and it looks ok. I think this should now be merged and someone with time should further look into completing the documentation.

**Review by bastelfreak — APPROVED** [review]
I  guess we should merge that now and drop centos6 in another PR

**Comment by genebean** [comment]
Old stuff was recently dropped - do you all want to merge as is or rebase and let test on w/o CentOS 6 and with Puppet 7?

**Comment by bastelfreak** [comment]
I'm going to merge it to finally have this in master.

**Pull request merged**